config/weapons.go

```diff
@@ -85,36 +85,79 @@ var WeaponTitles = []string{
 	"Grandmaster",
 }
 
+var AffinityTitles = []string{
+	"Unattuned",
+	"Neophyte",
+	"Novice",
+	"Channeler",
+	"Artisan",
+	"Specialist",
+	"Attuned",
+	"Elementalist",
+	"Savant",
+	"Virtuoso",
+	"Ascended",
+}
+
 func WeaponExpTitle(exp int, class int) string {
```

Comment:
I'm guessing you're over loading this because you expected mages to just never use weapons again.   Copy the function out and use AffinityTitles instead of checking the class and replacing it.